command/agent/config.go

```diff
@@ -638,6 +648,24 @@ func DecodeConfig(r io.Reader) (*Config, error) {
 		result.SessionTTLMin = dur
 	}
 
+	if result.AdvertiseAddrs.SerfLanRaw != "" {
+		addr, err := net.ResolveTCPAddr("tcp", result.AdvertiseAddrs.SerfLanRaw)
+		if err != nil {
+			return nil, fmt.Errorf("AdvertiseAddrs.SerfLan is invalid: %v", err)
+		} else {
```

Comment:
Minor nit-pick but let's drop these `else` statements, since we already return in the error path above.